mujoco_warp/_src/render_context.py

```diff
@@ -0,0 +1,248 @@
+  tex_data: wp.array(dtype=wp.uint32)
+  tex_height: wp.array(dtype=int)
+  tex_width: wp.array(dtype=int)
+  bvh_id: wp.uint64
+  mesh_bvh_ids: wp.array(dtype=wp.uint64)
+  lowers: wp.array(dtype=wp.vec3)
+  uppers: wp.array(dtype=wp.vec3)
+  groups: wp.array(dtype=wp.int32)
+  group_roots: wp.array(dtype=wp.int32)
+  pixels: wp.array3d(dtype=wp.uint32)
+  depth: wp.array3d(dtype=wp.float32)
+
+  def __init__(
+    self,
+    mjm: mujoco.MjModel,
+    m: Model,
+    d: Data,
+    nworld,
+    width,
+    height,
+    use_textures,
+    use_shadows,
+    render_rgb,
+    render_depth,
+    enabled_geom_groups = [0, 1, 2],
+  ):
+    
+    nmesh = mjm.nmesh
+    geom_enabled_idx = [i for i in range(mjm.ngeom) if mjm.geom_group[i] in enabled_geom_groups]
+    
+    used_mesh_ids = set(
+      int(mjm.geom_dataid[g])
+      for g in geom_enabled_idx
+      if mjm.geom_type[g] == GeomType.MESH and int(mjm.geom_dataid[g]) >= 0
+    )
+
+    self.mesh_registry = {}
+    mesh_bvh_ids = [wp.uint64(0) for _ in range(nmesh)]
+    mesh_bounds_size = [wp.vec3(0.0, 0.0, 0.0) for _ in range(nmesh)]
+
+    for i in range(nmesh):
+      if i not in used_mesh_ids:
+        continue
+
+      v_start = mjm.mesh_vertadr[i]
+      v_end = v_start + mjm.mesh_vertnum[i]
+      points = mjm.mesh_vert[v_start:v_end]
+
+      f_start = mjm.mesh_faceadr[i]
+      f_end = mjm.mesh_face.shape[0] if (i + 1) >= nmesh else mjm.mesh_faceadr[i + 1]
+      indices = mjm.mesh_face[f_start:f_end]
+      indices = indices.flatten()
+
+      mesh = wp.Mesh(
+        points=wp.array(points, dtype=wp.vec3),
+        indices=wp.array(indices, dtype=wp.int32),
+        bvh_constructor="sah",
+      )
+      self.mesh_registry[mesh.id] = mesh
+      mesh_bvh_ids[i] = mesh.id
+
+      pmin = points.min(axis=0)
+      pmax = points.max(axis=0)
+      half = 0.5 * (pmax - pmin)
+      mesh_bounds_size[i] = half
+    
+    tex_data_packed, tex_adr_packed = _create_packed_texture_data(mjm)
+
+    bvh_ngeom = len(geom_enabled_idx)
+    self.bvh_ngeom=bvh_ngeom
+    self.nworld=nworld
+    self.ncam=mjm.ncam
+    self.width=width
+    self.height=height
+    self.use_textures=use_textures
+    self.use_shadows=use_shadows
+    self.fov_rad=wp.radians(mjm.cam_fovy[0])
+    self.render_rgb=render_rgb
+    self.render_depth=render_depth
+    self.enabled_geom_ids=wp.array(geom_enabled_idx, dtype=int)
+    self.mesh_bvh_ids=wp.array(mesh_bvh_ids, dtype=wp.uint64)
+    self.mesh_bounds_size=wp.array(mesh_bounds_size, dtype=wp.vec3)
+    self.mesh_texcoord=wp.array(mjm.mesh_texcoord, dtype=wp.vec2)
+    self.mesh_texcoord_offsets=wp.array(mjm.mesh_texcoordadr, dtype=int)
+    self.mesh_texcoord_num=wp.array(mjm.mesh_texcoordnum, dtype=int)
+    self.tex_adr=tex_adr_packed
+    self.tex_data=tex_data_packed
+    self.tex_height=wp.array(mjm.tex_height, dtype=int)
+    self.tex_width=wp.array(mjm.tex_width, dtype=int)
+    self.lowers = wp.zeros((nworld * bvh_ngeom,), dtype=wp.vec3)
```

Comment:
`(nworld * bvh_ngeom,)` -> `nworld * bvh_ngeom`

there are similar instances we can update below